python/Makefile

```diff
@@ -17,13 +17,22 @@
 
 install:
 	pip install poetry
-	poetry install -E pyarrow
+	poetry install -E pyarrow -E s3fs
 
 lint:
 	poetry run pre-commit run --all-files
-
 test:
+	poetry run coverage run --source=pyiceberg/ -m pytest tests/ -m "not s3" ${PYTEST_ARGS}
+	poetry run coverage report -m --fail-under=90
+	poetry run coverage html
+	poetry run coverage xml
+test-all:
 	poetry run coverage run --source=pyiceberg/ -m pytest tests/ ${PYTEST_ARGS}
 	poetry run coverage report -m --fail-under=90
 	poetry run coverage html
 	poetry run coverage xml
+test-s3:
```

Comment:
Eventually I think we can maybe use some kind of arguments to generify but this is fine imo for now. And maybe later too depending.

Reply:
That's a great idea, we already use `PYTEST_ARGS` and I think we could just pass in the marked tests we want to run using that 👍